src/Authoring/WinRT.SourceGenerator/Generator.cs

```diff
@@ -28,91 +31,46 @@ internal sealed class WriteOnlyArrayAttribute : global::System.Attribute
     }
 }";
 
-        private string _tempFolder;
-
-        private static string GetAssemblyName(GeneratorExecutionContext context)
-        {
-            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.AssemblyName", out var assemblyName);
-            return assemblyName;
-        }
-
-        private static string GetAssemblyVersion(GeneratorExecutionContext context)
-        {
-            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.AssemblyVersion", out var assemblyVersion);
-            return assemblyVersion;
-        }
-
-        internal static string GetGeneratedFilesDir(GeneratorExecutionContext context)
-        {
-            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.CsWinRTGeneratedFilesDir", out var generatedFilesDir);
-            Directory.CreateDirectory(generatedFilesDir);
-            return generatedFilesDir;
-        }
-
-        private static bool IsCsWinRTComponent(GeneratorExecutionContext context)
-        {
-            if (context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.CsWinRTComponent", out var isCsWinRTComponentStr))
-            {
-                return bool.TryParse(isCsWinRTComponentStr, out var isCsWinRTComponent) && isCsWinRTComponent;
-            }
+        private Logger Logger { get; }
+        private readonly GeneratorExecutionContext context;
```

Comment:
We can still report diagnostics on a `readonly` context?

Reply:
The value we got was always readonly so we should still be able to report diagnostics without issues.